packages/dds/tree/src/simple-tree/schemaFactoryRecursive.ts

```diff
@@ -342,7 +237,53 @@ export class SchemaFactoryRecursive<
 					[string, InsertableTreeNodeFromImplicitAllowedTypesUnsafe<T>]
 				>;
 			},
-			false
+			false,
+			T
 		>;
 	}
 }
+
+/**
+ * Compile time check for validity of a recursive schema.
+ *
+ * @example
+ * ```typescript
+ * 	class Test extends sf.arrayRecursive("Test", [() => Test]) {}
```

Comment:
Nit: Probably don't need the extra indentation in the example.

Reply:
Or are you referring to that extra space. I hate trying to do indent in examples. SPace or tabs, its a mess either way...

Reply:
silly mixed space and tab mess there should be fixed.

Reply:
Yeah, I was just referring to the extra space.